custom/icds_reports/utils/aggregation_helpers/birth_preparedness_forms.py

```diff
@@ -71,31 +78,40 @@ def aggregation_query(self):
+          counsel_accessible_ppfp, ifa_last_seven_days,using_ifa
         ) (
           SELECT
             %(state_id)s AS state_id,
             %(month)s AS month,
-            ucr.case_id AS case_id,
-            ucr.latest_time_end AS latest_time_end_processed,
+            COALESCE(ucr.case_id, prev_month.case_id) AS case_id,
+            COALESCE(ucr.latest_time_end, prev_month.latest_time_end_processed) AS latest_time_end_processed,
             GREATEST(ucr.immediate_breastfeeding, prev_month.immediate_breastfeeding) AS immediate_breastfeeding,
-            ucr.anemia AS anemia,
-            ucr.eating_extra AS eating_extra,
-            ucr.resting AS resting,
-            ucr.anc_weight anc_weight,
-            ucr.anc_blood_pressure as anc_blood_pressure,
-            ucr.bp_sys as bp_sys,
-            ucr.bp_dia as bp_dia,
-            ucr.anc_hemoglobin as anc_hemoglobin,
-            ucr.bleeding as bleeding,
-            ucr.swelling as swelling,
-            ucr.blurred_vision as blurred_vision,
-            ucr.convulsions as convulsions,
-            ucr.rupture as rupture,
-            ucr.anc_abnormalities as anc_abnormalities,
-            COALESCE(ucr.valid_visits, 0) as valid_visits
+            COALESCE(ucr.anemia,prev_month.anemia) AS anemia,
+            COALESCE(ucr.eating_extra,prev_month.eating_extra) AS eating_extra,
+            COALESCE(ucr.resting,prev_month.resting) AS resting,
+            COALESCE(ucr.anc_weight,prev_month.anc_weight) anc_weight,
+            COALESCE(ucr.anc_blood_pressure,prev_month.anc_blood_pressure) as anc_blood_pressure,
+            COALESCE(ucr.bp_sys,prev_month.bp_sys) as bp_sys,
+            COALESCE(ucr.bp_dia,prev_month.bp_dia) as bp_dia,
+            COALESCE(ucr.anc_hemoglobin,prev_month.anc_hemoglobin) as anc_hemoglobin,
+            COALESCE(ucr.bleeding,prev_month.bleeding) as bleeding,
+            COALESCE(ucr.swelling,prev_month.swelling) as swelling,
+            COALESCE(ucr.blurred_vision,prev_month.blurred_vision) as blurred_vision,
+            COALESCE(ucr.convulsions,prev_month.convulsions) as convulsions,
+            COALESCE(ucr.rupture,prev_month.rupture) as rupture,
+            COALESCE(ucr.anc_abnormalities,prev_month.anc_abnormalities) as anc_abnormalities,
+            COALESCE(ucr.valid_visits, 0) as valid_visits,
+            GREATEST(ucr.play_birth_preparedness_vid, prev_month.play_birth_preparedness_vid) as play_birth_preparedness_vid,
+            GREATEST(ucr.counsel_preparation, prev_month.counsel_preparation) as counsel_preparation,
+            GREATEST(ucr.play_family_planning_vid, prev_month.play_family_planning_vid) as play_family_planning_vid,
+            GREATEST(ucr.conceive,prev_month.conceive) as conceive,
+            GREATEST(ucr.counsel_accessible_ppfp, prev_month.counsel_accessible_ppfp) as counsel_accessible_ppfp,
+            COALESCE(ucr.ifa_last_seven_days, prev_month.ifa_last_seven_days) as ifa_last_seven_days,
```

Comment:
using the previous month's `ifa_last_seven_days` seems wrong because it is definitely not the last 7 days. is that what we are doing wrong?

Reply:
hey I did this acc to the definition. Which says latest BP form submitted before the end of month. does not it makes sense to include the previous month ?
https://docs.google.com/spreadsheets/d/1RDLNIHF3iRkiQtWyrQjKH6f1bStF_S_9woXeITiMJKg/edit#gid=551271206&range=B40